File: 905-sort-array-by-parity/905-sort-array-by-parity.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& nums) 
    {
        vector<int>odd,even;
        for(int i{};i<nums.size();i++)
        {
            if(nums[i]&1)
            {
                odd.push_back(nums[i]);
            }
            else
            {
                even.push_back(nums[i]);
            }
            
        }
        vector<int>ans;
        for(int i:even)
        {
            ans.push_back(i);
        }
        for(int i:odd)
        {
            ans.push_back(i);
        }
        return ans;
        
        
    }
};

File: 905-sort-array-by-parity/905-sort-array-by-parity.java
class Solution {
    
    public void swap(int a,int b)
    {
        a=a-b;
        b=a+b;
        a=b-a;
    }
    public int[] sortArrayByParity(int[] nums) 
    {
        int i=0;
        int j=nums.length-1;
        while(i<j)
        {
            if(nums[i]%2==1 && nums[j]%2==0)
            {
                
                nums[i]=nums[i]-nums[j];
                nums[j]=nums[j]+nums[i];
                nums[i]=nums[j]-nums[i];
                i++;
                j--;
            }
            else if(nums[i]%2==0)
            {
                i++;
            }
            else if(nums[j]%2==1)
            {
                j--;
            }
        }
        return nums;
        
    }
}